src/main/java/core/basesyntax/strategy/PurchaseOperation.java

```diff
@@ -0,0 +1,11 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction, Map<String, Integer> storage) {
+        storage.merge(transaction.getFruit(), -transaction.getQuantity(), Integer::sum);
+    }
```

Comment:
Handling Purchase operation.
Check result balance before saving it in the Storage - it should be positive. Throw a RuntimeException in case the balance is negative. Currently, your `handle` method does not perform this check, which might result in negative balances in the storage.